### DISCUSSION TITLE
Dynamically sized UIcontainer

### DISCUSSION THREAD

**Discussion by MaelkMaelk** [MyreMylar/pygame_gui#458] [Q&A] [answered]
Is there a way to dynamically size a UIcontainer according to what is inside it, similar to how a UIbutton does it ?
I'm trying to do it this way:

cont = pygame_gui.core.ui_container.UIContainer(pygame.Rect((0,0),(-1, -1)), manager = manager)

Then I create my buttons with their containers set as this one.
But I end up with a rectangle with that is basically 0*0.
Am I doing something wrong ?

**Comment by hk0thari** [comment, marked as answer]
Hi MaelkMaelk,
I am just a contributor to this project, but I had started work on a similar container with a pull request which is yet to be accepted. You can either wait for that to be accepted or copy the code from the pull request labeled UIAutoResizingContainer or similar.

**Reply by MaelkMaelk** [reply]
Thank you very much!!! I had made my own bit of code for this issue but yours is immensely more functionnal. Thanks
